ui/src/message_popup/playtest/hmtca_experience_page.jsx

```diff
@@ -0,0 +1,203 @@
+/* @flow weak */
+import React from 'react';
+import uuid from 'uuid';
+import _ from 'lodash';
+
+import * as Api from '../../helpers/api.js';
+import hash from '../../helpers/hash.js';
+import LinearSession from '../linear_session/linear_session.jsx';
+import SessionFrame from '../linear_session/session_frame.jsx';
+import VelocityTransitionGroup from "velocity-react/velocity-transition-group";
+import 'velocity-animate/velocity.ui';
+import RaisedButton from 'material-ui/RaisedButton';
+import TextField from 'material-ui/TextField';
+import {RadioButton, RadioButtonGroup} from 'material-ui/RadioButton';
+
+import QuestionInterpreter from '../renderers/question_interpreter.jsx';
+import type {QuestionT} from './pairs_scenario.jsx';
+import HMTCAScenarios from './hmtca_scenario.jsx';
+import HMTCAGroupReview from './hmtca_group_review.jsx';
+
+type ResponseT = {
+  choice:string,
+  question:QuestionT,
+  sceneNumber?:number,
+  anonymizedText?:string
+};
+
+
+
+// This is the flow for the HMTCA breakout session
+export default React.createClass({
+  displayName: 'HMTCAExperiencePage',
+
+  propTypes: {
+    query: React.PropTypes.shape({
+      cohort: React.PropTypes.string,
+      p: React.PropTypes.string
+    }).isRequired
+  },
+
+  // User types cohort for team
+  getInitialState() {
+    return {
+      cohortKey: this.props.query.cohort || '',
+      bucketId: HMTCAScenarios.BUCKETS[0].id.toString(),
+      questions: null,
+      sessionId: uuid.v4()
+    };
+  },
+
+  onCohortKeyChanged(e) {
+    this.setState({ cohortKey: e.target.value });
+  },
+
+  onBucketChanged(e) {
+    this.setState({ bucketId: e.target.value });
+  },
+
+  // Making questions from the cohort
+  onStart(e) {
+    e.preventDefault();
+    const {cohortKey, bucketId} = this.state;
+    const allQuestions = HMTCAScenarios.questionsFor(cohortKey, parseInt(bucketId, 10));
+
+    const startQuestionIndex = this.props.query.p || 0; // for testing or demoing
+    const questions = allQuestions.slice(startQuestionIndex);
+    const questionsHash = hash(JSON.stringify(questions));
+    this.setState({
+      questions,
+      questionsHash
+    });
+  },
+
+  onResetSession() {
+    this.setState(this.getInitialState());
+  },
+
+  onLogMessage(type, response) {
+    const {cohortKey, bucketId, sessionId, questionsHash} = this.state;
+    
```

Comment:
This listens for particular log messages, and then writes them to two different places on the server - one the usual place with all the sessionId and timestamp data, and another with just the info about scenes and anonymized text responses.